### PR TITLE
Only inherit the handles that are really needed.

### PR DESCRIPTION
**Pull request by autoantwort**
Currently all handles are inherited to child processes. For more information see https://discord.com/channels/400588936151433218/687365466422902841/1072485165491814430

### CODE REVIEW

File: src/vcpkg/base/system.process.cpp [hunk 1]
```diff
@@ -693,35 +694,52 @@ namespace vcpkg
...
         }
 
-        // Ensure the write handle to the pipe for STDIN is not inherited.
-        if (!SetHandleInformation(ret.child_stdin, HANDLE_FLAG_INHERIT, 0))
+        startup_info_ex.StartupInfo.hStdError = startup_info_ex.StartupInfo.hStdOutput;
+
+        // Ensure that only the write handle to STDOUT and the read handle to STDIN are inherited.
+        // from https://devblogs.microsoft.com/oldnewthing/20111216-00/?p=8873
+        SIZE_T size = 0;
+        if (InitializeProcThreadAttributeList(nullptr, 1, 0, &size) || GetLastError() != ERROR_INSUFFICIENT_BUFFER)
```
Comment (BillyONeal) [inline src/vcpkg/base/system.process.cpp:713]: I would really like to see an RAII type for this.

Comment (autoantwort) [reply src/vcpkg/base/system.process.cpp:713]: Only for the clean up or for everything? So should a failure in the constructor (calls InitializeProcThreadAttributeList) throw an exception?

Comment (BillyONeal) [reply src/vcpkg/base/system.process.cpp:713]: I wish it were for the pipes too but that's preexisting so won't ask you to go there. I meant something like:

```
struct ProcAttributeList {
    ExpectedL<ProcAttributeList> create(DWORD dwAttributeCount);
    ExpectedL<Unit> update_attribute(DWORD_PTR Attribute, PVOID lpValue, SIZE_T cbSize);
    ~ProcAttributeList();

    ProcAttributeList(const ProcAttributeList&) = delete;
    ProcAttributeList& operator=(const ProcAttributeList&) = delete;
private:
    ProcAttributeList(/* *** */);
    std::vector<unsigned char> buffer;
};
```

Comment (autoantwort) [reply src/vcpkg/base/system.process.cpp]: Have done this in https://github.com/microsoft/vcpkg-tool/pull/887/commits/a3bbb3090d127f896cc0f2b03acac12b7c2b9e50

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**